
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>transposonmapper.processing.genomicfeatures_dataframe &#8212; SATAY pipeline at Delft :)</title>
    
  <link href="../../../_static/css/theme.css" rel="stylesheet" />
  <link href="../../../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../../../" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/jquery.js"></script>
    <script src="../../../_static/underscore.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/togglebutton.js"></script>
    <script src="../../../_static/clipboard.min.js"></script>
    <script src="../../../_static/copybutton.js"></script>
    <script src="../../../_static/tabs.js"></script>
    <script src="../../../_static/tabs.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <link rel="shortcut icon" href="../../../_static/genetica_modified_neon.png"/>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../../../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../../../_static/genetica_modified_neon.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">SATAY pipeline at Delft :)</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../Introduction.html">
   Status of the project
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Getting started
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../../../03-docker-doc/00-Docker-Intro.html">
   SATAY pipeline
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../03-docker-doc/01-Docker-Setup.html">
     Setting up docker
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../03-docker-doc/02-Docker-adapters.html">
     Creating a custom
     <em>
      adapters.fa
     </em>
     file
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../03-docker-doc/03-Docker-Developers.html">
     Building the Docker image
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../../getting-started/00-getting-started.html">
   Installing transposonmapper
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../../../06-Tutorials/00-name.html">
   Tutorials
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../06-Tutorials/tutorial.html">
     Transposonmapper output data postprocessing
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  About SATAY
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../../../00-Intro/what_is_satay.html">
   What is SATAY
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../00-Intro/experimental_setup.html">
     SATAY Experimental Process
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../00-Intro/gene_essentiality.html">
     Gene Essentiality in budding yeast
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../00-Intro/research-context-behind-satay.html">
     Research questions behind the use of SATAY
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  About Transposonmapper
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../../../00-Intro/flowchart.html">
   What is Transposonmapper
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../00-Intro/software-detailed-description-satay.sh.html">
     Overview of the software required for satay.sh
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../00-Intro/satay.sh-detailed-description-file-types.html">
     Detailed description of the output files from satay.sh
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../00-Intro/interpreting_insertions_reads.html">
     Interpreting Transposon Counts &amp; Reads
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Community
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../../../04-Developers/00-contributing-guidelines.html">
   Contributing guidelines for developers
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../../../05-Code-of-conduct/00-code-of-conduct.html">
   Code of Conduct
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  API reference
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../../../api/api.html">
   Transposonmapper
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../api/api_exporting.html">
     transposonmapper.exporting
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../api/api_importing.html">
     transposonmapper.importing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../api/api_mapping.html">
     transposonmapper.mapping
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../api/api_plotting.html">
     transposonmapper.plotting
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../api/api_processing.html">
     transposonmapper.processing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../api/api_properties.html">
     transposonmapper.properties
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../../api/api_statistics.html">
     transposonmapper.statistics
    </a>
   </li>
  </ul>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/SATAY-LL/Transposonmapper"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/SATAY-LL/Transposonmapper/issues/new?title=Issue%20on%20page%20%2F_modules/transposonmapper/processing/genomicfeatures_dataframe.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <h1>Source code for transposonmapper.processing.genomicfeatures_dataframe</h1><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">os</span><span class="o">,</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">matplotlib.patches</span> <span class="k">as</span> <span class="nn">mpatches</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span> 
<span class="kn">import</span> <span class="nn">pkg_resources</span>


<span class="kn">from</span> <span class="nn">transposonmapper.properties</span> <span class="kn">import</span> <span class="n">chromosome_position</span><span class="p">,</span><span class="n">gene_aliases</span>
<span class="kn">from</span> <span class="nn">transposonmapper.processing</span> <span class="kn">import</span> <span class="n">chromosome_name_wigfile</span>

<span class="kn">from</span> <span class="nn">transposonmapper.processing.read_sgdfeatures</span> <span class="kn">import</span> <span class="n">sgd_features</span>


<span class="kn">from</span> <span class="nn">transposonmapper.importing</span> <span class="kn">import</span> <span class="p">(</span>
    <span class="n">load_default_files</span><span class="p">,</span> <span class="n">load_sgd_tab</span>
<span class="p">)</span>


<span class="kn">from</span> <span class="nn">transposonmapper.processing.dna_features_helpers</span> <span class="kn">import</span> <span class="p">(</span><span class="n">build_dataframe</span><span class="p">,</span> <span class="n">input_region</span><span class="p">,</span> <span class="n">intergenic_regions</span><span class="p">,</span> <span class="n">read_pergene_file</span><span class="p">,</span> 
                                                              <span class="n">read_wig_file</span><span class="p">,</span><span class="n">gene_location</span><span class="p">,</span><span class="n">checking_features</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">transposonmapper.plotting.dna_features_plot</span> <span class="kn">import</span> <span class="n">dna_features_plot</span>

<div class="viewcode-block" id="dna_features"><a class="viewcode-back" href="../../../api/api_processing.html#transposonmapper.processing.dna_features">[docs]</a><span class="k">def</span> <span class="nf">dna_features</span><span class="p">(</span><span class="n">region</span><span class="p">,</span> <span class="n">wig_file</span><span class="p">,</span> <span class="n">pergene_insertions_file</span><span class="p">,</span> <span class="n">variable</span><span class="o">=</span><span class="s2">&quot;reads&quot;</span><span class="p">,</span> <span class="n">plotting</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">savefigure</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">True</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;This scripts takes a user defined genomic region (i.e. chromosome number, region or gene) and creates a dataframe including information about all genomic features in the chromosome (i.e. genes, nc-DNA etc.).</span>
<span class="sd">    This can be used to determine the number of reads outside the genes to use this for normalization of the number of reads in the genes.</span>
<span class="sd">    Output is a dataframe including major information about all genomic features and optionally a barplot indicating the number of transposons per genomic region.</span>
<span class="sd">    A genomic region is here defined as a gene (separated as annotated essential and not essential), telomere, centromere, ars etc.</span>
<span class="sd">    This can be used for identifying neutral regions (i.e. genomic regions that, if inhibited, do not influence the fitness of the cells).</span>
<span class="sd">    This function can be used for normalizing the transposon insertions per gene using the neutral regions.</span>
<span class="sd">    </span>
<span class="sd">              </span>
<span class="sd">    </span>
<span class="sd">    Parameters</span>
<span class="sd">    ----------</span>
<span class="sd">    region : str</span>
<span class="sd">        - Region: e.g. chromosome number (either a normal number between 1 and 16 or in roman numerals between I and XVI), a list like [&#39;V&#39;, 0, 14790] which creates a barplot between basepair 0 and 14790) or a genename.</span>

<span class="sd">    wig_file : str</span>
<span class="sd">        absolute path for the wig file location</span>
<span class="sd">    pergene_insertions_file : str </span>
<span class="sd">        asbsoulte path for the _pergene_insertions.txt file location </span>
<span class="sd">    variable : str, optional</span>
<span class="sd">        By default &quot;reads&quot;. It could be &quot;transposons&quot;or &quot;reads&quot;. This would be used for the plotting if True </span>
<span class="sd">    plotting : bool, optional</span>
<span class="sd">        Whether or not producing a bar plot with the reads/insertions per genomic location in the region, by default True</span>
<span class="sd">    savefigure : bool, optional</span>
<span class="sd">        Whether or not saving the plot in the same folder as the datafiles, by default False</span>
<span class="sd">    verbose : bool, optional</span>
<span class="sd">        Determines how much textual feedback is given. When set to False, only warnings will be shown. By default True</span>

<span class="sd">    Returns</span>
<span class="sd">    -------</span>
<span class="sd">    dataframe</span>
<span class="sd">         Dataframe containing information about the selected chromosome. </span>

<span class="sd">    </span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="c1"># If necessary, load default files</span>
    <span class="n">gff_file</span><span class="p">,</span> <span class="n">essentials_file</span><span class="p">,</span> <span class="n">gene_information_file</span> <span class="o">=</span> <span class="n">load_default_files</span><span class="p">()</span>
    <span class="n">sgd_features_file</span><span class="o">=</span><span class="n">load_sgd_tab</span><span class="p">()</span>

    <span class="c1"># Verify presence of files</span>
    <span class="n">data_files</span> <span class="o">=</span> <span class="p">{</span>
        <span class="s2">&quot;gff3&quot;</span><span class="p">:</span> <span class="n">gff_file</span><span class="p">,</span>
        <span class="s2">&quot;essentials&quot;</span><span class="p">:</span> <span class="n">essentials_file</span><span class="p">,</span>
        <span class="s2">&quot;gene_names&quot;</span><span class="p">:</span> <span class="n">gene_information_file</span><span class="p">,</span>
        <span class="s2">&quot;sgd_features&quot;</span><span class="p">:</span> <span class="n">sgd_features_file</span>
    <span class="p">}</span>

    <span class="k">for</span> <span class="n">filetype</span><span class="p">,</span> <span class="n">file_path</span> <span class="ow">in</span> <span class="n">data_files</span><span class="o">.</span><span class="n">items</span><span class="p">():</span>
        <span class="k">assert</span> <span class="n">file_path</span><span class="p">,</span> <span class="sa">f</span><span class="s2">&quot;</span><span class="si">{</span><span class="n">filetype</span><span class="si">}</span><span class="s2"> not found at </span><span class="si">{</span><span class="n">file_path</span><span class="si">}</span><span class="s2">&quot;</span>


    <span class="n">variable</span> <span class="o">=</span> <span class="n">variable</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span>
    <span class="k">if</span> <span class="n">plotting</span> <span class="o">==</span> <span class="kc">True</span><span class="p">:</span>
        <span class="n">create_plottitle</span> <span class="o">=</span> <span class="s1">&#39;&#39;</span>

<span class="c1"># DETERMINE INPUTTED REGION</span>

    <span class="n">roi_start</span><span class="p">,</span><span class="n">roi_end</span><span class="p">,</span><span class="n">region_type</span><span class="p">,</span><span class="n">chrom</span><span class="o">=</span><span class="n">input_region</span><span class="p">(</span><span class="n">region</span><span class="o">=</span><span class="n">region</span><span class="p">,</span><span class="n">verbose</span><span class="o">=</span><span class="n">verbose</span><span class="p">)</span>

    

<span class="c1">#READ WIG FILE FOR GETTING LOCATIONS OF ALL TN INSERTIONS</span>

    <span class="n">insrt_in_chrom_list</span><span class="p">,</span><span class="n">reads_in_chrom_list</span><span class="o">=</span><span class="n">read_wig_file</span><span class="p">(</span><span class="n">wig_file</span><span class="o">=</span><span class="n">wig_file</span><span class="p">,</span><span class="n">chrom</span><span class="o">=</span><span class="n">chrom</span><span class="p">)</span>


<span class="c1"># READ PERGENE_INSERTIONS FILE FOR LOCATION OF ALL INSERTIONS PER EACH GENE.</span>

    <span class="n">gene_position_dict</span><span class="o">=</span><span class="n">read_pergene_file</span><span class="p">(</span><span class="n">pergene_insertions_file</span><span class="o">=</span><span class="n">pergene_insertions_file</span><span class="p">,</span><span class="n">chrom</span><span class="o">=</span><span class="n">chrom</span><span class="p">)</span>

<span class="c1"># DETERMINE THE LOCATION GENOMIC FEATURES IN THE CURRENT CHROMOSOME AND STORE THIS IN A DICTIONARY</span>

    <span class="n">dna_dict</span><span class="p">,</span><span class="n">start_chr</span><span class="p">,</span><span class="n">end_chr</span><span class="p">,</span><span class="n">len_chr</span><span class="p">,</span><span class="n">feature_orf_dict</span><span class="o">=</span><span class="n">gene_location</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span><span class="n">gene_position_dict</span><span class="p">,</span><span class="n">verbose</span><span class="p">)</span>

<span class="c1">## GET FEATURES FROM INTERGENIC REGIONS </span>

    <span class="n">dna_dict_new</span><span class="p">,</span><span class="n">genomicregions_list</span><span class="o">=</span><span class="n">intergenic_regions</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span><span class="n">start_chr</span><span class="p">,</span><span class="n">dna_dict</span><span class="p">)</span>


    <span class="c1">### TEST IF ELEMENTS IN FEATURE_ORF_DICT FOR SELECTED CHROMOSOME ARE THE SAME AS THE GENES IN GENE_POSITION_DICT BY CREATING THE DICTIONARY FEATURE_POSITION_DICT CONTAINING ALL THE GENES IN FEATURE_ORF_DICT WITH THEIR CORRESPONDING POSITION IN THE CHROMOSOME</span>
    <span class="n">checking_features</span><span class="p">(</span><span class="n">feature_orf_dict</span><span class="p">,</span><span class="n">chrom</span><span class="p">,</span><span class="n">gene_position_dict</span><span class="p">,</span><span class="n">verbose</span><span class="p">)</span>

    <span class="n">dna_df2</span><span class="o">=</span><span class="n">build_dataframe</span><span class="p">(</span><span class="n">dna_dict_new</span><span class="p">,</span><span class="n">start_chr</span><span class="p">,</span><span class="n">end_chr</span><span class="p">,</span><span class="n">insrt_in_chrom_list</span><span class="p">,</span><span class="n">reads_in_chrom_list</span><span class="p">,</span><span class="n">genomicregions_list</span><span class="p">,</span><span class="n">chrom</span><span class="p">)</span>

    <span class="c1">#PRINT INFORMATION FOR THE SELECTED GENE</span>
    <span class="k">if</span> <span class="n">region_type</span> <span class="o">==</span> <span class="s1">&#39;Gene&#39;</span><span class="p">:</span>
        <span class="k">for</span> <span class="n">region_info</span> <span class="ow">in</span> <span class="n">dna_df2</span><span class="o">.</span><span class="n">itertuples</span><span class="p">():</span>
            <span class="k">if</span> <span class="n">region_info</span><span class="o">.</span><span class="n">Feature_name</span> <span class="o">==</span> <span class="n">region</span><span class="o">.</span><span class="n">upper</span><span class="p">()</span> <span class="ow">or</span> <span class="n">region_info</span><span class="o">.</span><span class="n">Standard_name</span> <span class="o">==</span> <span class="n">region</span><span class="o">.</span><span class="n">upper</span><span class="p">():</span>
                <span class="nb">print</span><span class="p">(</span><span class="n">region_info</span><span class="p">)</span>

    
    <span class="c1"># CREATE BAR PLOT </span>
    <span class="n">dna_features_plot</span><span class="p">(</span><span class="n">chrom</span><span class="p">,</span> <span class="n">dna_df2</span><span class="p">,</span><span class="n">roi_start</span><span class="p">,</span><span class="n">roi_end</span><span class="p">,</span><span class="n">len_chr</span><span class="p">,</span><span class="n">plotting</span><span class="p">,</span><span class="n">variable</span><span class="p">)</span>

    <span class="k">if</span> <span class="n">savefigure</span> <span class="o">==</span> <span class="kc">True</span><span class="p">:</span>
        <span class="n">file_dirname</span><span class="o">=</span><span class="n">pkg_resources</span><span class="o">.</span><span class="n">resource_filename</span><span class="p">(</span><span class="s2">&quot;transposonmapper&quot;</span><span class="p">,</span> <span class="s2">&quot;data_files/&quot;</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">variable</span> <span class="o">==</span> <span class="s1">&#39;reads&#39;</span><span class="p">:</span>
            <span class="n">saving_name</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">file_dirname</span><span class="p">,</span><span class="s1">&#39;GenomicFeaturesReads_Barplot_Chrom&#39;</span><span class="o">+</span><span class="n">chrom</span><span class="o">+</span><span class="s1">&#39;_NonNormalized&#39;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">saving_name</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">file_dirname</span><span class="p">,</span><span class="s1">&#39;GenomicFeaturesInsertions_Barplot_Chrom&#39;</span><span class="o">+</span><span class="n">chrom</span><span class="o">+</span><span class="s1">&#39;_NonNormalized&#39;</span><span class="p">)</span>
        <span class="n">plt</span><span class="o">.</span><span class="n">savefig</span><span class="p">(</span><span class="n">saving_name</span><span class="p">,</span> <span class="n">orientation</span><span class="o">=</span><span class="s1">&#39;landscape&#39;</span><span class="p">,</span> <span class="n">dpi</span><span class="o">=</span><span class="mi">200</span><span class="p">)</span>
        <span class="n">plt</span><span class="o">.</span><span class="n">close</span><span class="p">()</span>

    <span class="k">return</span><span class="p">(</span><span class="n">dna_df2</span><span class="p">)</span></div>






</pre></div>

              </div>
              
        
            



<div class='prev-next-bottom'>
    

</div>
        
        </div>
    </div>
    <footer class="footer">
    <div class="container">
      <p>
        
          By Gregory van Beek , Maurits Kok, Leila IÃ±igo de la Cruz<br/>
        
            &copy; Copyright Apache 2.0.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../../../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>