
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Transposonmapper output data postprocessing &#8212; SATAY pipeline at Delft :)</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet" />
  <link href="../_static/css/index.c5995385ac14fb8791e8eb36b4908be2.css" rel="stylesheet" />

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.acff12b8f9c144ce68a297486a2fa670.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="../_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.1c5a1a01449ed65a7b51.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/tabs.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"TeX": {"Macros": {"N": "\\mathbb{N}", "floor": ["\\lfloor#1\\rfloor", 1], "bmat": ["\\left[\\begin{array}"], "emat": ["\\end{array}\\right]"]}}, "tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <link rel="shortcut icon" href="../_static/genetica_modified_neon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="What is SATAY" href="../00-Intro/what_is_satay.html" />
    <link rel="prev" title="Tutorials" href="00-name.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      <img src="../_static/genetica_modified_neon.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">SATAY pipeline at Delft :)</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../Introduction.html">
   Status of the project
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Getting started
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../03-docker-doc/00-Docker-Intro.html">
   SATAY pipeline
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../03-docker-doc/01-Docker-Setup.html">
     Setting up docker
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../03-docker-doc/02-Docker-adapters.html">
     Creating a custom
     <em>
      adapters.fa
     </em>
     file
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../03-docker-doc/03-Docker-Developers.html">
     Building the Docker image
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../getting-started/00-getting-started.html">
   Installing transposonmapper
  </a>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="00-name.html">
   Tutorials
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     Transposonmapper output data postprocessing
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  About SATAY
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../00-Intro/what_is_satay.html">
   What is SATAY
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/experimental_setup.html">
     SATAY Experimental Process
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/gene_essentiality.html">
     Gene Essentiality in budding yeast
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/research-context-behind-satay.html">
     Research questions behind the use of SATAY
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  About Transposonmapper
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../00-Intro/flowchart.html">
   What is Transposonmapper
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/software-detailed-description-satay.sh.html">
     Overview of the software required for satay.sh
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/satay.sh-detailed-description-file-types.html">
     Detailed description of the output files from satay.sh
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/interpreting_insertions_reads.html">
     Interpreting Transposon Counts &amp; Reads
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Community
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../04-Developers/00-contributing-guidelines.html">
   Contributing guidelines for developers
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../05-Code-of-conduct/00-code-of-conduct.html">
   Code of Conduct
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  API reference
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../api/api.html">
   Transposonmapper
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../api/api_exporting.html">
     transposonmapper.exporting
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../api/api_importing.html">
     transposonmapper.importing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../api/api_mapping.html">
     transposonmapper.mapping
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../api/api_plotting.html">
     transposonmapper.plotting
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../api/api_processing.html">
     transposonmapper.processing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../api/api_properties.html">
     transposonmapper.properties
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../api/api_statistics.html">
     transposonmapper.statistics
    </a>
   </li>
  </ul>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/06-Tutorials/tutorial.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/SATAY-LL/Transposonmapper"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/SATAY-LL/Transposonmapper/issues/new?title=Issue%20on%20page%20%2F06-Tutorials/tutorial.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org//v2/gh/SATAY-LL/Transposonmapper/gh-pages?urlpath=lab/tree/jupyter_book/_build/html/06-Tutorials/tutorial.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Transposonmapper output data postprocessing
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#how-to-clean-the-wig-and-bed-files">
   How to clean the wig and bed files
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#visualize-the-insertions-and-reads-per-gene-throughout-the-genome">
   Visualize the insertions and reads per gene  throughout the genome
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#zoom-in-into-the-chromosomes">
   Zoom in into the chromosomes
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#volcano-plots">
   Volcano plots
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#getting-the-volcano-plot">
     Getting the volcano plot
    </a>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#this-is-a-volcano-plot-made-with-real-data">
     This is a volcano plot made with real data!
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="transposonmapper-output-data-postprocessing">
<h1>Transposonmapper output data postprocessing<a class="headerlink" href="#transposonmapper-output-data-postprocessing" title="Permalink to this headline">¶</a></h1>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">## Importing the required python libraries </span>
<span class="kn">import</span> <span class="nn">os</span><span class="o">,</span> <span class="nn">sys</span>
<span class="kn">import</span> <span class="nn">warnings</span>
<span class="kn">import</span> <span class="nn">timeit</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span> 
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="how-to-clean-the-wig-and-bed-files">
<h1>How to clean the wig and bed files<a class="headerlink" href="#how-to-clean-the-wig-and-bed-files" title="Permalink to this headline">¶</a></h1>
<p>Here we will remove transposon insertions in .bed and .wig files that were mapped outside the chromosomes, creates consistent naming for chromosomes and change the header of files with custom headers.</p>
<p>Clean wig files for proper visualization in the genome Browser <a class="reference external" href="http://genome-euro.ucsc.edu/cgi-bin/hgGateway">http://genome-euro.ucsc.edu/cgi-bin/hgGateway</a></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">transposonmapper.processing.clean_bedwigfiles</span> <span class="kn">import</span> <span class="n">cleanfiles</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">######## Lets save the wig and bed files as variables to clean them and call the function#####################</span>

<span class="n">wig_files</span><span class="o">=</span><span class="p">[]</span>
<span class="n">bed_files</span><span class="o">=</span><span class="p">[]</span>

<span class="n">data_dir</span><span class="o">=</span><span class="s2">&quot;../transposonmapper/data_files/files4test/&quot;</span>
<span class="k">for</span> <span class="n">root</span><span class="p">,</span> <span class="n">dirs</span><span class="p">,</span> <span class="n">files</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">walk</span><span class="p">(</span><span class="n">data_dir</span><span class="p">):</span>
    <span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">files</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;sorted.bam.wig&quot;</span><span class="p">):</span>
            <span class="n">wig_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">root</span><span class="p">,</span> <span class="n">file</span><span class="p">))</span>
        <span class="k">elif</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;sorted.bam.bed&quot;</span><span class="p">):</span>
             <span class="n">bed_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">root</span><span class="p">,</span> <span class="n">file</span><span class="p">))</span>

<span class="c1">############## Cleaning the files #############################</span>
            
<span class="n">custom_header</span> <span class="o">=</span> <span class="s2">&quot;&quot;</span>
<span class="n">split_chromosomes</span> <span class="o">=</span> <span class="kc">False</span>
<span class="k">for</span> <span class="n">files</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">wig_files</span><span class="p">,</span><span class="n">bed_files</span><span class="p">):</span>
    <span class="n">cleanfiles</span><span class="p">(</span><span class="n">filepath</span><span class="o">=</span><span class="n">files</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">custom_header</span><span class="o">=</span><span class="n">custom_header</span><span class="p">,</span> <span class="n">split_chromosomes</span><span class="o">=</span><span class="n">split_chromosomes</span><span class="p">)</span>
    <span class="n">cleanfiles</span><span class="p">(</span><span class="n">filepath</span><span class="o">=</span><span class="n">files</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">custom_header</span><span class="o">=</span><span class="n">custom_header</span><span class="p">,</span> <span class="n">split_chromosomes</span><span class="o">=</span><span class="n">split_chromosomes</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="visualize-the-insertions-and-reads-per-gene-throughout-the-genome">
<h1>Visualize the insertions and reads per gene  throughout the genome<a class="headerlink" href="#visualize-the-insertions-and-reads-per-gene-throughout-the-genome" title="Permalink to this headline">¶</a></h1>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span> <span class="c1">## Import the function</span>
<span class="kn">from</span> <span class="nn">transposonmapper.processing.transposonread_profileplot_genome</span> <span class="kn">import</span> <span class="n">profile_genome</span>


<span class="c1">####Lets save the cleaned files as variables to clean them and call the function####</span>
<span class="n">cleanbed_files</span><span class="o">=</span><span class="p">[]</span>
<span class="k">for</span> <span class="n">root</span><span class="p">,</span> <span class="n">dirs</span><span class="p">,</span> <span class="n">files</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">walk</span><span class="p">(</span><span class="n">data_dir</span><span class="p">):</span>
    <span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">files</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;clean.bed&quot;</span><span class="p">):</span>
            <span class="n">cleanbed_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">root</span><span class="p">,</span> <span class="n">file</span><span class="p">))</span>

<span class="n">cleanwig_files</span><span class="o">=</span><span class="p">[]</span>
<span class="k">for</span> <span class="n">root</span><span class="p">,</span> <span class="n">dirs</span><span class="p">,</span> <span class="n">files</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">walk</span><span class="p">(</span><span class="n">data_dir</span><span class="p">):</span>
    <span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">files</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;clean.wig&quot;</span><span class="p">):</span>
            <span class="n">cleanwig_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">root</span><span class="p">,</span> <span class="n">file</span><span class="p">))</span>


<span class="c1">#### vizualization #####</span>
<span class="n">bed_file</span><span class="o">=</span><span class="n">cleanbed_files</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="c1"># example for the 1st file </span>
<span class="n">variable</span><span class="o">=</span><span class="s2">&quot;transposons&quot;</span> <span class="c1">#&quot;reads&quot; &quot;transposons&quot;</span>
<span class="n">bar_width</span><span class="o">=</span><span class="kc">None</span>
<span class="n">savefig</span><span class="o">=</span><span class="kc">False</span>

<span class="n">profile</span><span class="o">=</span><span class="n">profile_genome</span><span class="p">(</span><span class="n">bed_file</span><span class="o">=</span><span class="n">bed_file</span><span class="p">,</span> <span class="n">variable</span><span class="o">=</span><span class="n">variable</span><span class="p">,</span> <span class="n">bar_width</span><span class="o">=</span><span class="n">bar_width</span><span class="p">,</span> <span class="n">savefig</span><span class="o">=</span><span class="n">savefig</span><span class="p">,</span><span class="n">showfig</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p><img alt="" src="../_images/profile-plot-dummy.png" /></p>
</div>
<div class="section" id="zoom-in-into-the-chromosomes">
<h1>Zoom in into the chromosomes<a class="headerlink" href="#zoom-in-into-the-chromosomes" title="Permalink to this headline">¶</a></h1>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">transposonmapper.processing.genomicfeatures_dataframe</span> <span class="kn">import</span> <span class="n">dna_features</span>

<span class="c1">##### getting the files #########</span>
<span class="n">pergene_files</span><span class="o">=</span><span class="p">[]</span>

<span class="n">data_dir</span><span class="o">=</span><span class="s2">&quot;../transposonmapper/data_files/files4test/&quot;</span>
<span class="k">for</span> <span class="n">root</span><span class="p">,</span> <span class="n">dirs</span><span class="p">,</span> <span class="n">files</span> <span class="ow">in</span> <span class="n">os</span><span class="o">.</span><span class="n">walk</span><span class="p">(</span><span class="n">data_dir</span><span class="p">):</span>
    <span class="k">for</span> <span class="n">file</span> <span class="ow">in</span> <span class="n">files</span><span class="p">:</span>
        <span class="k">if</span> <span class="n">file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s1">&#39;sorted.bam_pergene_insertions.txt&#39;</span><span class="p">):</span>
            <span class="n">pergene_files</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">root</span><span class="p">,</span> <span class="n">file</span><span class="p">))</span>


<span class="c1">#### vizualization #####</span>


<span class="n">wig_file</span> <span class="o">=</span> <span class="n">cleanwig_files</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">pergene_insertions_file</span> <span class="o">=</span> <span class="n">pergene_files</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">plotting</span><span class="o">=</span><span class="kc">True</span>
<span class="n">variable</span><span class="o">=</span><span class="s2">&quot;reads&quot;</span> <span class="c1">#&quot;reads&quot; or &quot;insertions&quot;</span>
<span class="n">savefigure</span><span class="o">=</span><span class="kc">False</span>
<span class="n">verbose</span><span class="o">=</span><span class="kc">True</span>

   
<span class="n">region</span> <span class="o">=</span> <span class="s2">&quot;I&quot;</span> <span class="c1">#e.g. 1, &quot;I&quot;, [&quot;I&quot;, 0, 10000&quot;], gene name (e.g. &quot;CDC42&quot;)</span>
<span class="n">dna_features</span><span class="p">(</span><span class="n">region</span><span class="o">=</span><span class="n">region</span><span class="p">,</span>
                <span class="n">wig_file</span><span class="o">=</span><span class="n">wig_file</span><span class="p">,</span>
                <span class="n">pergene_insertions_file</span><span class="o">=</span><span class="n">pergene_insertions_file</span><span class="p">,</span>
                <span class="n">variable</span><span class="o">=</span><span class="n">variable</span><span class="p">,</span>
                <span class="n">plotting</span><span class="o">=</span><span class="n">plotting</span><span class="p">,</span>
                <span class="n">savefigure</span><span class="o">=</span><span class="n">savefigure</span><span class="p">,</span>
                <span class="n">verbose</span><span class="o">=</span><span class="n">verbose</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>This is the plot for the case of the dummy sample files for chromosome I.</p>
<p><img alt="" src="../_images/region_I_dna_features_dummy.png" /></p>
</div>
<div class="section" id="volcano-plots">
<h1>Volcano plots<a class="headerlink" href="#volcano-plots" title="Permalink to this headline">¶</a></h1>
<p>Do you want to compare two differente libraries to discover which genes stood out from their comparison?</p>
<p>Then do volcano plots!!</p>
<div class="section" id="getting-the-volcano-plot">
<h2>Getting the volcano plot<a class="headerlink" href="#getting-the-volcano-plot" title="Permalink to this headline">¶</a></h2>
<p>Look at the help of this function , <a class="reference external" href="https://github.com/SATAY-LL/Transposonmapper/blob/main/transposonmapper/statistics/volcanoplot.py">HERE</a></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">transposonmapper.statistics</span> <span class="kn">import</span> <span class="n">volcano</span>

<span class="n">path_a</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;&quot;</span>
<span class="n">filelist_a</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;&quot;</span><span class="p">,</span><span class="s2">&quot;&quot;</span><span class="p">]</span>
<span class="n">path_b</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;&quot;</span>
<span class="n">filelist_b</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;&quot;</span><span class="p">,</span><span class="s2">&quot;&quot;</span><span class="p">]</span>


<span class="n">variable</span> <span class="o">=</span> <span class="s1">&#39;read_per_gene&#39;</span> <span class="c1">#&#39;read_per_gene&#39; &#39;tn_per_gene&#39;, &#39;Nreadsperinsrt&#39;</span>
<span class="n">significance_threshold</span> <span class="o">=</span> <span class="mf">0.01</span> <span class="c1">#set threshold above which p-values are regarded significant</span>
<span class="n">normalize</span><span class="o">=</span><span class="kc">True</span>

<span class="n">trackgene_list</span> <span class="o">=</span> <span class="p">[</span><span class="s1">&#39;my-favorite-gene&#39;</span><span class="p">]</span> <span class="c1"># [&quot;cdc42&quot;]</span>


<span class="n">figure_title</span> <span class="o">=</span> <span class="s2">&quot; &quot;</span>

<span class="n">volcano_df</span> <span class="o">=</span> <span class="n">volcano</span><span class="p">(</span><span class="n">path_a</span><span class="o">=</span><span class="n">path_a</span><span class="p">,</span> <span class="n">filelist_a</span><span class="o">=</span><span class="n">filelist_a</span><span class="p">,</span>
            <span class="n">path_b</span><span class="o">=</span><span class="n">path_b</span><span class="p">,</span> <span class="n">filelist_b</span><span class="o">=</span><span class="n">filelist_b</span><span class="p">,</span>
            <span class="n">variable</span><span class="o">=</span><span class="n">variable</span><span class="p">,</span>
            <span class="n">significance_threshold</span><span class="o">=</span><span class="n">significance_threshold</span><span class="p">,</span>
            <span class="n">normalize</span><span class="o">=</span><span class="n">normalize</span><span class="p">,</span>
            <span class="n">trackgene_list</span><span class="o">=</span><span class="n">trackgene_list</span><span class="p">,</span>
            <span class="n">figure_title</span><span class="o">=</span><span class="n">figure_title</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="this-is-a-volcano-plot-made-with-real-data">
<h2>This is a volcano plot made with real data!<a class="headerlink" href="#this-is-a-volcano-plot-made-with-real-data" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>Comparing the libraries of wild type vs <span class="math notranslate nohighlight">\(\Delta\)</span> nrp1</p></li>
</ul>
<p><img alt="" src="../_images/example-with-real-data-volcano.png" /></p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./06-Tutorials"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="00-name.html" title="previous page">Tutorials</a>
    <a class='right-next' id="next-link" href="../00-Intro/what_is_satay.html" title="next page">What is SATAY</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Gregory van Beek , Maurits Kok, Leila Iñigo de la Cruz<br/>
        
            &copy; Copyright Apache 2.0.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.1c5a1a01449ed65a7b51.js"></script>

  
  </body>
</html>