
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>transposonmapper.statistics &#8212; SATAY pipeline at Delft :)</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="../_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/tabs.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.d59cb220de22ca1c485ebbdc042f0030.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/genetica_modified_neon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="transposonmapper.properties" href="api_properties.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/genetica_modified_neon.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">SATAY pipeline at Delft :)</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../Introduction.html">
   Status of the project
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Getting started
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../07-preprocessing-steps/preprocessing-steps.html">
   Preprocessing steps
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../03-docker-doc/00-Docker-Intro.html">
   SATAY pipeline
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../03-docker-doc/01-Docker-Setup.html">
     Setting up docker
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../03-docker-doc/02-Docker-adapters.html">
     Creating a custom
     <em>
      adapters.fa
     </em>
     file
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../03-docker-doc/03-Docker-Developers.html">
     Building the Docker image
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../getting-started/00-getting-started.html">
   Installing transposonmapper
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../06-Tutorials/00-name.html">
   Tutorials
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../06-Tutorials/tutorial.html">
     Transposonmapper output data postprocessing
    </a>
   </li>
  </ul>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  About SATAY
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../00-Intro/what_is_satay.html">
   What is SATAY
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/experimental_setup.html">
     SATAY Experimental Process
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/gene_essentiality.html">
     Gene Essentiality in budding yeast
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/research-context-behind-satay.html">
     Research questions behind the use of SATAY
    </a>
   </li>
  </ul>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  About Transposonmapper
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../00-Intro/flowchart.html">
   What is Transposonmapper
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/software-detailed-description-satay.sh.html">
     Overview of the software required for satay.sh
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/satay.sh-detailed-description-file-types.html">
     Detailed description of the output files from satay.sh
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../00-Intro/interpreting_insertions_reads.html">
     Interpreting Transposon Counts &amp; Reads
    </a>
   </li>
  </ul>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  Community
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../04-Developers/00-contributing-guidelines.html">
   Contributing guidelines for developers
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../05-Code-of-conduct/00-code-of-conduct.html">
   Code of Conduct
  </a>
 </li>
</ul>
<p aria-level="2" class="caption" role="heading">
 <span class="caption-text">
  API reference
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="api.html">
   Transposonmapper
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="api_exporting.html">
     transposonmapper.exporting
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="api_importing.html">
     transposonmapper.importing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="api_mapping.html">
     transposonmapper.mapping
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="api_plotting.html">
     transposonmapper.plotting
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="api_processing.html">
     transposonmapper.processing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="api_properties.html">
     transposonmapper.properties
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     transposonmapper.statistics
    </a>
   </li>
  </ul>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../_sources/api/api_statistics.rst"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.rst</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
                onclick="printPdf(this)" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/SATAY-LL/Transposonmapper"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/SATAY-LL/Transposonmapper/issues/new?title=Issue%20on%20page%20%2Fapi/api_statistics.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show noprint">
            
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>transposonmapper.statistics</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                    </div>
                </div>
            </div>
            
              <div>
                
  <div class="section" id="module-transposonmapper.statistics">
<span id="transposonmapper-statistics"></span><h1>transposonmapper.statistics<a class="headerlink" href="#module-transposonmapper.statistics" title="Permalink to this headline">¶</a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="transposonmapper.statistics.apply_stats">
<span class="sig-prename descclassname"><span class="pre">transposonmapper.statistics.</span></span><span class="sig-name descname"><span class="pre">apply_stats</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">variable_a_array</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variable_b_array</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">significance_threshold</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">volcano_df</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/transposonmapper/statistics/volcano_helpers.html#apply_stats"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#transposonmapper.statistics.apply_stats" title="Permalink to this definition">¶</a></dt>
<dd><p>This function computes the statistics measure for the volcano plot</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>variable_a_array</strong> (<em>array</em>) – The values (# of insertions or reads) of the replicates of one library</p></li>
<li><p><strong>variable_b_array</strong> (<em>array</em>) – The values (# of insertions or reads) of the  replicates of the other library</p></li>
<li><p><strong>significance_threshold</strong> (<em>float</em>) – It will use the default value in the volcano function which is 0.01</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A dataframe containing all the info for the volcano plot.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="transposonmapper.statistics.dataframe_from_pergenefile">
<span class="sig-prename descclassname"><span class="pre">transposonmapper.statistics.</span></span><span class="sig-name descname"><span class="pre">dataframe_from_pergenefile</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pergenefile</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/transposonmapper/statistics/dataframe_from_pergene.html#dataframe_from_pergenefile"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#transposonmapper.statistics.dataframe_from_pergenefile" title="Permalink to this definition">¶</a></dt>
<dd><p>This function creates a dataframe with the information from a pergene.txt file.</p>
<p>The gene_essentiality is created based on the genes present in the Cerevisiae_EssentialGenes_List_1.txt and Cerevisiae_EssentialGenes_List_2.txt files
The number of reads per insertion (Nreadsperinsrt) is determined by dividing the read_per_gene column by the tn_per_gene column.</p>
<p>Author: Gregory van Beek</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pergenefile</strong> (<em>str</em>) – absolute path to the pergene.txt file , one of the outputs of the transposonmapper module</p></li>
<li><p><strong>verbose</strong> (<em>bool</em><em>, </em><em>optional</em>) – [description], by default True</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><dl class="simple">
<dt>Output is a dataframe where each row is a single gene and with the following columns:</dt><dd><ul class="simple">
<li><p>gene_names</p></li>
<li><p>gene_essentiality</p></li>
<li><p>tn_per_gene</p></li>
<li><p>read_per_gene</p></li>
<li><p>Nreadsperinsrt</p></li>
</ul>
</dd>
</dl>
</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>dataframe</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="transposonmapper.statistics.essential_genes">
<span class="sig-prename descclassname"><span class="pre">transposonmapper.statistics.</span></span><span class="sig-name descname"><span class="pre">essential_genes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">genenames_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lines</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/transposonmapper/statistics/dataframe_from_pergene_helpers.html#essential_genes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#transposonmapper.statistics.essential_genes" title="Permalink to this definition">¶</a></dt>
<dd><p>It provides a list of essential genes</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>genenames_list</strong> (<em>list</em>) – A list will al genes names that were mapped to the reference genome</p></li>
<li><p><strong>lines</strong> (<em>int</em>) – Number of genes in total</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>List of essential genes</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>list</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="transposonmapper.statistics.info_from_datasets">
<span class="sig-prename descclassname"><span class="pre">transposonmapper.statistics.</span></span><span class="sig-name descname"><span class="pre">info_from_datasets</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">datafiles_list_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">datafiles_list_b</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variable</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">normalize</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/transposonmapper/statistics/volcano_helpers.html#info_from_datasets"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#transposonmapper.statistics.info_from_datasets" title="Permalink to this definition">¶</a></dt>
<dd><p>Read the information contain in the datafiles for the volcano plot</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>datafiles_list_a</strong> (<em>list of str</em>) – List of the absolute paths of all the replicates from the
reference library.</p></li>
<li><p><strong>datafiles_list_b</strong> (<em>list of str</em>) – List of the absolute paths of all the replicates from the
experimental  library.</p></li>
<li><p><strong>variable</strong> (<em>str</em>) – Magnitude indicating based on what to make the volcano plot.
For example: tn_per_gene, read_per_gene or Nreadsperinsrt</p></li>
<li><p><strong>normalize</strong> (<em>bool</em>) – If True , If set to True, each gene is normalized based on
the total count in each dataset (i.e. each file in <a href="#id1"><span class="problematic" id="id2">filelist_</span></a>)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>variable_a_array</strong> (<em>numpy.array</em>)</p></li>
<li><p><strong>variable_b_array</strong> (<em>numpy.array</em>)</p></li>
<li><p><strong>volcano_df</strong> (<em>pandas.core.frame.DataFrame</em>)</p></li>
<li><p><strong>tnread_gene_a</strong> (<em>pandas.core.frame.DataFrame</em>)</p></li>
<li><p><strong>tnread_gene_b</strong> (<em>pandas.core.frame.DataFrame</em>)</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="transposonmapper.statistics.make_datafile">
<span class="sig-prename descclassname"><span class="pre">transposonmapper.statistics.</span></span><span class="sig-name descname"><span class="pre">make_datafile</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filelist_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_b</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filelist_b</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/transposonmapper/statistics/volcano_helpers.html#make_datafile"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#transposonmapper.statistics.make_datafile" title="Permalink to this definition">¶</a></dt>
<dd><p>Assembly the datafile name to analyze</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>path_a</strong> (<em>str</em>) – Path of the files corresponding to the reference library</p></li>
<li><p><strong>filelist_a</strong> (<em>list of str</em>) – List of the filenames of the different replicates from the reference library.
It has to have minimum two replicates per library, so the list has to contain
a minimum of two files.</p></li>
<li><p><strong>path_b</strong> (<em>str</em>) – Path of the files corresponding to the experimental library</p></li>
<li><p><strong>filelist_b</strong> (<em>list of str</em>) – List of the filenames of the different replicates from the experimental library.
It has to have minimum two replicates per library, so the list has to contain
a minimum of two files.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Complete paths of the reference and the experimental libraries</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="transposonmapper.statistics.read_pergene_file">
<span class="sig-prename descclassname"><span class="pre">transposonmapper.statistics.</span></span><span class="sig-name descname"><span class="pre">read_pergene_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pergenefile</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/transposonmapper/statistics/dataframe_from_pergene_helpers.html#read_pergene_file"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#transposonmapper.statistics.read_pergene_file" title="Permalink to this definition">¶</a></dt>
<dd><p>It reads the content of the pergene file , one of the outputs of Transposonmapper</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pergenefile</strong> (<em>str</em>) – absolute path to the pergene.txt file , one of the outputs of the transposonmapper module</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><ul class="simple">
<li><p><em>list</em> – Gene names list</p></li>
<li><p><em>list</em> – Insertion list</p></li>
<li><p><em>list</em> – Reads list</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="transposonmapper.statistics.reads_per_insertion">
<span class="sig-prename descclassname"><span class="pre">transposonmapper.statistics.</span></span><span class="sig-name descname"><span class="pre">reads_per_insertion</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">tnpergene_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">readpergene_list</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">lines</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/transposonmapper/statistics/dataframe_from_pergene_helpers.html#reads_per_insertion"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#transposonmapper.statistics.reads_per_insertion" title="Permalink to this definition">¶</a></dt>
<dd><p>It computes the reads per insertion following the formula:
reads/(insertions-1) if the number of insertions is higher than 5,
if not then the reads per insertion will be 0.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>tnpergene_list</strong> (<em>list</em>) – A list with all insertions</p></li>
<li><p><strong>readpergene_list</strong> (<em>list</em>) – A list of the reads</p></li>
<li><p><strong>lines</strong> (<em>int</em>) – Number of genes mapped to in the reference genome</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A list containing all the reads per insertions per gene.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>list</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="transposonmapper.statistics.volcano">
<span class="sig-prename descclassname"><span class="pre">transposonmapper.statistics.</span></span><span class="sig-name descname"><span class="pre">volcano</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filelist_a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path_b</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filelist_b</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fold_change_interval</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p_value_interval</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">variable</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'read_per_gene'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">significance_threshold</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.01</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">normalize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">trackgene_list</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figure_title</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">savefigure</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/transposonmapper/statistics/volcanoplot.html#volcano"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#transposonmapper.statistics.volcano" title="Permalink to this definition">¶</a></dt>
<dd><p>This script creates a volcanoplot to show the significance of fold change between two datasets.
It is based on this website:</p>
<blockquote>
<div><ul class="simple">
<li><p><a class="reference external" href="https://towardsdatascience.com/inferential-statistics-series-t-test-using-numpy-2718f8f9bf2f">https://towardsdatascience.com/inferential-statistics-series-t-test-using-numpy-2718f8f9bf2f</a></p></li>
<li><p><a class="reference external" href="https://www.statisticshowto.com/independent-samples-t-test/">https://www.statisticshowto.com/independent-samples-t-test/</a></p></li>
</ul>
</div></blockquote>
<dl class="simple">
<dt>Code for showing gene name when hovering over datapoint is based on:</dt><dd><ul class="simple">
<li><p><a class="reference external" href="https://stackoverflow.com/questions/7908636/possible-to-make-labels-appear-when-hovering-over-a-point-in-matplotlib">https://stackoverflow.com/questions/7908636/possible-to-make-labels-appear-when-hovering-over-a-point-in-matplotlib</a></p></li>
</ul>
</dd>
</dl>
<p>T-test is measuring the number of standard deviations our measured mean is from the baseline mean, while taking into
account that the standard deviation of the mean can change as we get more data
This creates a volcano plot that shows the fold change between two libraries and the corresponding p-values.</p>
<p>The fold change is determined by the mean of dataset b (experimental set) divided by the mean of dataset a (reference set).
The datasets can be of different length.
P-value is determined based on the student t-test (scipy.stats.ttest_ind).</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The fold change is determined by the ratio between the reference and the experimental dataset.
When one of the datasets is 0, this is false results for the fold change.
To prevent this, the genes with 0 insertions are set to have 5 insertions, and the genes with 0 reads are set to have 25 reads.
These values are determined in dicussion with the Kornmann lab.</p>
</div>
<ul class="simple">
<li><p>Created on Tue Feb 16 14:06:48 2021</p></li>
<li><p>&#64;author: gregoryvanbeek</p></li>
</ul>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>path_a</strong> (<em>str</em>) – paths to location of the datafiles for library a</p></li>
<li><p><strong>filelist_a</strong> (<em>str</em>) – list of the names of the datafiles for library a  located in path_a.
The type of file here is the pergene.txt file , which is one of the outputs from  the transposonmapper function.
The format of the pergene file should be TAB separated and NOT COMMA separated. if you have
it as comma separated you can convert to tab separated using the command line with
this command: <cite>cat oldfile.txt | tr ‘[,]’ ‘[    ]’ &gt; newfile.txt</cite></p></li>
<li><p><strong>path_b</strong> (<em>str</em>) – paths to location of the datafiles for library b</p></li>
<li><p><strong>filelist_b</strong> (<em>str</em>) – list of the names of the datafiles for  library b located in path_b
The type of file here is the pergene.txt file , which is one of the outputs from  the transposonmapper function.
The format of the pergene file should be TAB separated and NOT COMMA separated. if you have
it as comma separated you can convert to tab separated using the command line with
this command: <cite>cat oldfile.txt | tr ‘[,]’ ‘[    ]’ &gt; newfile.txt</cite></p></li>
<li><p><strong>fold_change_interval</strong> (<em>numpy.array</em>) – a vector of two elements where the first element(positive value) set the upper threshold for the positive
values and the second element(negative value) set the lower threshold for the negative values.</p></li>
</ul>
</dd>
</dl>
<dl>
<dt>p_value_interval<span class="classifier">numpy.array</span></dt><dd><blockquote>
<div><p>a vector of two elements where the first element set the upper threshold for the positive</p>
</div></blockquote>
<p>values and the second element set the lower threshold for the negative values.</p>
</dd>
<dt>variable<span class="classifier">str, optional</span></dt><dd><p>tn_per_gene, read_per_gene or Nreadsperinsrt , by default ‘read_per_gene’</p>
</dd>
<dt>significance_threshold<span class="classifier">float, optional</span></dt><dd><p>Threshold value above which the fold change is regarded significant, only for plotting, by default 0.01</p>
</dd>
<dt>normalize<span class="classifier">bool, optional</span></dt><dd><p>Whether to normalize variable. If set to True, each gene is normalized based on the total count in each dataset (i.e. each file in <a href="#id3"><span class="problematic" id="id4">filelist_</span></a>)
, by default True</p>
</dd>
<dt>trackgene_list<span class="classifier">list, optional</span></dt><dd><p>Enter a list of gene name(s) which will be highlighted in the plot (e.g. [‘cdc42’, ‘nrp1’]), by default []</p>
</dd>
<dt>figure_title<span class="classifier">str, optional</span></dt><dd><p>The title of the figure if not empty, by default “”</p>
</dd>
</dl>
<p>savefigure : bool, optional</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p><ul>
<li><p><em>dataframe</em> –</p>
<p>A dataframe containing:</p>
<blockquote>
<div><ul class="simple">
<li><p>gene_names</p></li>
<li><p>fold change</p></li>
<li><p>t statistic</p></li>
<li><p>p value</p></li>
<li><p>whether p value is above threshold</p></li>
</ul>
</div></blockquote>
</li>
<li><p><em>figure</em> –</p>
<ul class="simple">
<li><p>volcanoplot with the log2 fold change between the two libraries and the -log10 p-value.</p></li>
</ul>
</li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./api"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            
                <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="api_properties.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">transposonmapper.properties</p>
        </div>
    </a>
</div>
            
        </div>
    </div>
    <footer class="footer">
  <p>
    
      By Gregory van Beek , Maurits Kok, Leila Iñigo de la Cruz<br/>
    
        &copy; Copyright Apache 2.0.<br/>
  </p>
</footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>